File: data_classes/fooditem.py
from __future__ import annotations

class FoodItem:
    def __init__(
        self,
        meal: str,
        group: str,
        name: str,
        description: str,
        ingredients: list[str],
        allergens: list[str],
        calories: int,
        caloriesFromFat: int,
        fat: str,
        saturatedFat: str,
        transFat: str,
        polyunsaturatedFat: str,
        cholesterol: str,
        sodium: str,
        carbohydrates: str,
        dietaryFiber: str,
        sugar: str,
        protein: str,
        potassium: str,
        iron: str,
        calcium: str,
        vitaminA: str,
        vitaminC: str,
        vitaminD: str,
        portionSize: str,
        portion: str,
        isVegan: bool,
        isVegetarian: bool,
        isMindful: bool,
        isSwell: bool,
        isPlantBased: bool
    ):
        self.meal = meal
        self.group = group
        self.name = name
        self.description = description
        self.ingredients = ingredients
        self.allergens = allergens
        self.calories = calories
        self.caloriesFromFat = caloriesFromFat
        self.fat = fat
        self.saturatedFat = saturatedFat
        self.transFat = transFat
        self.polyunsaturatedFat = polyunsaturatedFat
        self.cholesterol = cholesterol
        self.sodium = sodium
        self.carbohydrates = carbohydrates
        self.dietaryFiber = dietaryFiber
        self.sugar = sugar
        self.protein = protein
        self.potassium = potassium
        self.iron = iron
        self.calcium = calcium
        self.vitaminA = vitaminA
        self.vitaminC = vitaminC
        self.vitaminD = vitaminD
        self.portionSize = portionSize
        self.portion = portion
        self.isVegan = isVegan
        self.isVegetarian = isVegetarian
        self.isMindful = isMindful
        self.isSwell = isSwell
        self.isPlantBased = isPlantBased
        self.reviews = []
        self.overallRating = 0.0
        self.todayRating = 0.0

File: tc-dine-eval-frontend/src/types/foodItem.ts
import type { Review } from "./review";

<<<<<<< HEAD
export type MealName = "BREAKFAST" | "LUNCH" | "AFTERNOON_SNACK" | "DINNER";

export interface FoodItem {
  id: string;
  name: string;
  meal: MealName;
  reviews?: Review[];
  activeRating: number;
  overallRating: number;
}
=======
export type MealType = "BREAKFAST" | "LUNCH" | "AFTERNOON_SNACK" | "DINNER";

export interface FoodItem {
  id: string;
  mealType: MealType;
  group: string;
  name: string;
  description: string;
  ingredients: string[];
  allergens: string[];

  calories: number;
  caloriesFromFat: number;
  fat: string;
  saturatedFat: string;
  transFat: string;
  polyunsaturatedFat: string;
  cholesterol: string;
  sodium: string;
  carbohydrates: string;
  dietaryFiber: string;
  sugar: string;
  protein: string;
  potassium: string;
  iron: string;
  calcium: string;
  vitaminA: string;
  vitaminC: string;
  vitaminD: string;

  portionSize: string;
  portion: string;

  isVegan: boolean;
  isVegetarian: boolean;
  isMindful: boolean;
  isSwell: boolean;
  isPlantBased: boolean;

  ratingCount: number;
  overallRating: number;
  todayRating: number;

  reviews?: Review[];
}



>>>>>>> d28bd9aea6d516f218774cdf28f516c9fca1d6cb
